dockerfiles/hpcc/templates/_helpers.tpl

```diff
@@ -47,9 +47,17 @@ Generate global ConfigMap info
 Pass in root as .
 */}}
 {{- define "hpcc.generateGlobalConfigMap" -}}
+{{- $defaultEsp := "" -}}
 imageVersion: {{ required "Please specify .global.image.version" .Values.global.image.version | quote }}
 singleNode: {{ .Values.global.singleNode | default false }}
-defaultEsp: {{ .Values.global.defaultEsp | default (index .Values.esp 0).name }}
+{{- if .Values.global.defaultEsp -}}
+ {{- $defaultEsp := .Values.global.defaultEsp -}}
+{{- else if hasPrefix "[]" (typeOf .Values.esp) -}}
+ {{- if (len .Values.esp) -}}
+  {{- $defaultEsp := (index .Values.esp 0).name -}}
```

Comment:
I don't think this works, because of the scoping rules. It will set $defaultEsp in the scope of just this if block.
The trick I've seen to get this to work is:
```
{{- if (len .Values.esp) -}}
  $dummy := set . "defaultEsp"  (index .Values.esp 0).name -}}
{{- end }}

...

defaultEsp: {{ .defaultEsp }}
```